osu.Game/Online/Leaderboards/LeaderboardScoresProvider.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Development;
+using osu.Framework.Graphics;
+using osu.Game.Online.API;
+
+namespace osu.Game.Online.Leaderboards
+{
+    public abstract partial class LeaderboardScoresProvider<TScope, TScoreInfo> : Component
```

Comment:
This looks like a very forced structure. Two generic type parameters, a whole bunch of members with no documentations that look very overfitted to the two implementations of this abstract that you have there.

I think this abstract class should not exist. I'm not sure this class is a good abstraction or even achieving much of code duplication reduction, it mainly looks to be obfuscating logic. Its responsibilities can and should be handled by its current derivative classes.